tests/testthat/test-organizations.R

```diff
@@ -1,3 +1,12 @@
+with_mock_dir("gh_organizations", {
```

Comment:
Rather than having two `with_mock_dir` examples, you could make this one use a mock that you aren't planning to re-record. IMO this is a good example of where tying the tests to real requests means you end up with (1) bloated mocks with more records/details than you need to test your R code, while also (2) tests assertions that are vague and don't clearly test that you are returning the expected value. 

Because the API response may contain different records each time you record them, you can't assert the values, only that the function returns a character vector. That could hide lots of bugs. But if you knew that the API response couldn't change (because you've frozen it by recording it once), you could `expect_identical(gh_organizations(), c("errfree", "engineyard", ...))`.

You could go as far as reducing the mock to

```
[
    {
        "login": "errfree"
    },
    {
        "login": "engineyard"
    }
]
```

and assert the exact expected return value.

Reply:
Right, I can edit that (I use a minimal response in the presser PR #4 for status ). But if you have many endpoints, isn't manually creating/editing mock fastidious? And how do you easily notice changes upstream? (When recording mocks you can re-record and look at the git diff for instance) reg my second question I suppose running all tests without mocks regularly would show bugs due to changes upstream but not the response structure change directly 🤔

Reply:
That said this discussion is good for the chapter where I will compare the three packages, so I could definitely show one test with a fake mock in this demo.

Reply:
right and you should choose based on your experience with the API (or APIs in general... like, if a few APIs changed without your noticing, then you might be warier in the future)

Reply:
I have used a simpler mock in the status test this way now the demo shows both things.


I still use with_mock_dir because it allows me to get a starting point to edit the first time I run it (I especially like that I then don't need to create the mock file at the right path by hand). Does this make sense to you? The demo should make the package look good. :-)